devito/builtins/initializers.py

```diff
@@ -62,6 +66,17 @@ def assign(f, rhs=0, options=None, name='assign', **kwargs):
     else:
         for i, j in zip(as_list(f), rhs):
             eqs.append(dv.Eq(i, j))
+
+    if not domain_only:
+        subs = {}
+        for d, h in zip(f.dimensions, f._size_halo):
+            if sum(h) == 0:
+                continue
+            subs[d] = dv.CustomDimension(name=d.name, parent=d,
```

Comment:
no, because remember those are global values, so it will break with MPI

Secondly, I wanted to abstract that complexity ("how much should x_m, x_M, y_m ... be?") away